libsplinter/src/biome/credentials/rest_resources.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use uuid::Uuid;
+
+use crate::actix_web::HttpResponse;
+use crate::futures::{Future, IntoFuture};
+use crate::rest_api::{into_bytes, ErrorResponse, Method, Resource};
+
+use super::super::users::{user_store::SplinterUserStore, SplinterUser, UserStore};
+use super::{
+    credentials_store::SplinterCredentialsStore, CredentialsStore, CredentialsStoreError,
+    UserCredentialsBuilder,
+};
+
+#[derive(Deserialize)]
+struct UsernamePassword {
+    username: String,
+    password: String,
+}
+
+pub fn make_register_route(
+    credentials_store: Arc<SplinterCredentialsStore>,
+    user_store: Arc<SplinterUserStore>,
+) -> Resource {
+    Resource::build("/biome/register").add_method(Method::Post, move |_, payload| {
```

Comment:
Add to spec

Reply:
I don't think I should add it to spec yet.. Because the API spec is in splinterd and currently this route is not being included in splinterd REST API. Which raises the question of how to documents these optional endpoints in libsplinter side. I guess doc comments is an option.